### PR TITLE
Coding conventions

### PR DESCRIPTION
**Pull request by abhinavachanta**
Files by abhinav:
test/src/receiver/main.cpp
test/src/sender/main.cpp
test/src/subnet/main.cpp

Files by Rajeev:
include/atomics/receiver_cadmium.hpp
include/atomics/sender_cadmium.hpp
include/atomics/subnet_cadmium.hpp

### CODE REVIEW

**Comment by syedomarazam** [comment]
Good coding conventions are not implemented properly. Please see all my comments and update the structure names, variable names accordingly in the code per good coding standards defined and discussed in class by professor.

When you break a line, make sure the alignment is proper and readable.

eg : cadmium in 2nd line must align with cadmium in first line after “=“ sign. 

This is applicable to all the places in code. So update all the places where you have implemented “Line < 80 characters” rule.

File: include/atomics/receiver_cadmium.hpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct Receiver_defs
```
Comment (syedomarazam) [inline include/atomics/receiver_cadmium.hpp]: it should be receiver_defs

all lower case letters.

File: include/atomics/subnet_cadmium.hpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct Subnet_defs
```
Comment (syedomarazam) [inline include/atomics/subnet_cadmium.hpp]: it should be subnet_defs

all lower case letters.

File: include/atomics/sender_cadmium.hpp [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct Sender_defs
```
Comment (syedomarazam) [inline include/atomics/sender_cadmium.hpp]: it should be sender_defs

all lower case letters.

**Review by syedomarazam — COMMENTED** [review]
all structures, the variable name must be lower case.

File: include/atomics/receiver_cadmium.hpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    struct in: public in_port <Message_t> {};
+};
+
+template <typename TIME>
+    class Receiver
+    {
+        using definitions = Receiver_defs; // putting definitions in context
+        public:
+            //Parameters to be overwriten when instantiating the atomic model
+            TIME preparation_time;
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp]: preparation_time is a constant name and should be UPPERCASE

File: include/atomics/receiver_cadmium.hpp [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        using output_ports = std::tuple <typename definitions::out> ;
+
+        // internal transition
+        void internal_transition()
+        {
+            state.sending = false;
+        }
+
+        // external transition
+        void external_transition(TIME e, typename make_message_bags
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp]: Alignment function as per Coding Convention for a function

File: include/atomics/receiver_cadmium.hpp [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            for (const auto &x: get_messages <typename definitions:: in> (mbs))
+            {
+                state.acknowledgement_number = static_cast <int> (x.value);
+                state.sending = true;
+            }
+
+        }
+
+        // confluence transition
+        void confluence_transition(TIME e, typename make_message_bags <input_ports>::type mbs)
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp]: greater than 80 lines. break the line

File: include/atomics/sender_cadmium.hpp [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+    struct ackReceivedOut: public out_port <Message_t> {};
+    struct dataOut: public out_port <Message_t> {};
+    struct controlIn: public in_port <Message_t> {};
+    struct ackIn: public in_port <Message_t> {};
+};
+
+template <typename TIME>
+    class Sender
+    {
+        using definitions = Sender_defs; // putting definitions in context
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: struct ion name should be lower case

**Comment by pcsmartapps** [comment]
Under atomics (receiver_cadmium.hpp, sender_cadmium.hpp and subnet_cadmium.hpp)
1- Please make sure the structure names follow coding conventions:
Ie, Message_t => should be message_t and so on.
2- function are broken and aligned properly
Example,
void confluence_transition(TIME e, 
.......................................................typename make_message_bags <input_ports>::type mbs)
3- General Lines are broken and aligned properly,
Ie,
friend std::ostringstream &operator << (std::ostringstream &os, 
..................................................................................const typename Receiver <TIME>::state_type &i)

Under test (include/atomics/receiver_cadmium.hpp, include/atomics/sender_cadmium.hpp, include/atomics/subnet_cadmium.hpp)
**make sure also the 3 points above are followed**
Another example:
using info =  cadmium::logger::logger<cadmium::logger::logger_info, 
...........................cadmium::dynamic::logger::formatter<TIME>,oss_sink_provider>;

**Review by pcsmartapps — CHANGES_REQUESTED** [review]

**Comment by abhinavachanta** [comment]
Code updated as per the review comments. Pls check

File: include/atomics/receiver_cadmium.hpp (hunk 4 above)
Comment (abhinavachanta) [reply include/atomics/receiver_cadmium.hpp]: Changed it to upper case: PREPARATION_TIME

File: include/atomics/receiver_cadmium.hpp (hunk 5 above)
Comment (pcsmartapps) [reply include/atomics/receiver_cadmium.hpp]: Alignment function as per Coding Convention for a function means the following
void external_transition(TIME e, typename make_message_bags 
................................................<input_ports>::type mbs)

OR

void external_transition(TIME e,
................................................typename make_message_bags<input_ports>::type mbs)

think of the dots as spaces

File: include/atomics/receiver_cadmium.hpp (hunk 4 above)
Comment (pcsmartapps) [reply include/atomics/receiver_cadmium.hpp]: preparation_time UPPERCASE is changed under receiver but not under sender.

File: include/atomics/sender_cadmium.hpp [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        using output_ports = std::tuple <typename definitions::packetSentOut, 
+																typename definitions::ackReceivedOut, typename definitions::dataOut> ;
+
+        // internal transition
+        void internal_transition()
+        {
+            if (state.ack)
+            {
+                if (state.packet_number < state.total_packet_number)
+                {
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: All if statement  needs to follow coding conventions as specified by the professor:
ie, Starting curly bracket at the end of the first line. The ending curly bracket should go on a new line in the same column where the if begins. 
if (value == 1) {
   value = 2;
}
pls, apply to all if statement code convention  in your code

File: include/atomics/sender_cadmium.hpp [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+                    else
+                    {
+                        if (state.next_internal != std::numeric_limits <TIME>::infinity())
+                        {
+                            state.next_internal = state.next_internal - e;
+                        }
+                    }
+                }
+            }
+            for (const auto &x: get_messages <typename definitions::ackIn> (mbs))
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: All for statement  needs to follow coding conventions as specified by professor:
ie,
for  () {
   value = 2;
}
pls, apply it through out your code

**Review by syedomarazam — CHANGES_REQUESTED** [review]
rename all your test files as
main_sender.cpp
main_receiver.cpp
main_subnet.cpp

it's difficult to do code review on GitHub because all the files have the same name. It creates confusion in readers minds whether he/she is looking at the correct file or not.

File: test/src/receiver/main.cpp [hunk 10]
```diff
@@ -12,118 +11,136 @@
...
-  ApplicationGen(const char* file_path) : iestream_input<Message_t,T>(file_path) {}
-};
-
-
-int main(){
+template <typename T>
+    class ApplicationGen: public iestream_input <message_t, T>
+    {
+        public: ApplicationGen() =
+            default;
```
Comment (syedomarazam) [inline test/src/receiver/main.cpp]: it must go to line 35.

public: ApplicationGen() = default;

File: include/atomics/sender_cadmium.hpp [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+            int packet_number;
+            int total_packet_number;
+            int alt_bit;
+            bool sending;
+            bool model_active;
+            TIME next_internal;
+        };
+        state_type state;
+        // ports definition
+        using input_ports = std::tuple <typename definitions::controlIn,
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: The lines 68 and 70 are too long and should be aligned properly

File: include/atomics/subnet_cadmium.hpp [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            state.transmiting = false;
+        }
+
+        // external transition
+        void external_transition(TIME e, 
+								 typename make_message_bags <input_ports>::type mbs)
+        {
+            state.index++;
+            if (get_messages <typename definitions:: in> (mbs).size() > 1)
```
Comment (pcsmartapps) [inline include/atomics/subnet_cadmium.hpp]: fix all if and for statements as pointed before

File: include/atomics/sender_cadmium.hpp [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+                    state.next_internal = PREPARATION_TIME;
+                }
+            }
+        }
+
+        // external transition
+        void external_transition(TIME e,
+								 typename make_message_bags <input_ports>::type mbs)
+        {
+            if ((get_messages <typename definitions::controlIn> (mbs).size() + get_messages
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: line is too long and  if should be changed accordingly

File: include/atomics/sender_cadmium.hpp [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+        // time_advance function
+        TIME time_advance() const
+        {
+            return state.next_internal;
+        }
+
+        friend std::ostringstream &operator << (std::ostringstream &os, 
+												const typename Sender <TIME>::state_type &i)
+        {
+            os << "packet_number: " << i.packet_number << " &total_packet_number: " << i.total_packet_number;
```
Comment (pcsmartapps) [inline include/atomics/sender_cadmium.hpp]: line is more than 80 character

File: test/src/receiver/main.cpp [hunk 15]
```diff
@@ -12,118 +11,136 @@
...
- oports_TOP, 
- eics_TOP, 
- eocs_TOP, 
- ics_TOP 
-  );
-
-///****************////
-
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
+    using info = cadmium::logger::logger <cadmium::logger::logger_info,
```
Comment (syedomarazam) [inline test/src/receiver/main.cpp]: alignment is not per good coding practices in all using statements (line 56 to 75). Looks like I'm climbing steps.

When you break a line make sure you do it like below.
eg: Line#64 (ignore dots)
    using routing = cadmium::logger::logger <cadmium::logger::logger_message_routing, 
    .......................................................................................cadmium::dynamic::logger::formatter <TIME>,
    .......................................................................................oss_sink_provider> ;

File: test/src/receiver/main.cpp [hunk 16]
```diff
@@ -12,118 +11,136 @@
...
+
+    /*******************************************/
+
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control = "test/data/receiver_input_test.txt"; 
+	const char *i_input_data_control = input_data_control.c_str();
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator =
```
Comment (syedomarazam) [inline test/src/receiver/main.cpp]: alignment issue

File: test/src/receiver/main.cpp [hunk 17]
```diff
@@ -12,118 +11,136 @@
...
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator = 
+														cadmium::dynamic::translate::make_dynamic_atomic_model 
+																								< ApplicationGen, TIME, const char *> ("generator", std::move
+																																					(i_input_data_control));
+
+    /********************************************/
+    /****** RECIEVER *******************/
+    /********************************************/
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> receiver1 =
```
Comment (syedomarazam) [inline test/src/receiver/main.cpp]: alignment issue

File: test/src/sender/main.cpp [hunk 18]
```diff
@@ -13,129 +13,151 @@
...
- oports_TOP, 
- eics_TOP, 
- eocs_TOP, 
- ics_TOP 
-  );
-
-///****************////
-
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
+    using info = cadmium::logger::logger <cadmium::logger::logger_info,
```
Comment (syedomarazam) [inline test/src/sender/main.cpp]: alignment issues in Line 59 to 77. fix it referring to receiver main.cpp comments.

File: test/src/sender/main.cpp [hunk 19]
```diff
@@ -13,129 +13,151 @@
...
+
+    /*******************************************/
+
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control = "test/data/sender_input_test_control_In.txt"; 
+	const char *i_input_data_control = input_data_control.c_str();
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_con =
```
Comment (syedomarazam) [inline test/src/sender/main.cpp]: alignment issue, not readable

File: test/src/sender/main.cpp [hunk 20]
```diff
@@ -13,129 +13,151 @@
...
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_con = 
+															cadmium::dynamic::translate::make_dynamic_atomic_model 
+																										< ApplicationGen, TIME, const char *> ("generator_con", 
+																																				std::move(i_input_data_control));
+
+    string input_data_ack = "test/data/sender_input_test_ack_In.txt"; 
+	const char *i_input_data_ack = input_data_ack.c_str();
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_ack =
```
Comment (syedomarazam) [inline test/src/sender/main.cpp]: alignment issue

File: test/src/sender/main.cpp [hunk 21]
```diff
@@ -13,129 +13,151 @@
...
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_ack = 
+															cadmium::dynamic::translate::make_dynamic_atomic_model 
+																										< ApplicationGen, TIME, const char *> ("generator_ack", std::move 
+																																								(i_input_data_ack));
+
+    /********************************************/
+    /****** SENDER *******************/
+    /********************************************/
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> sender1 =
```
Comment (syedomarazam) [inline test/src/sender/main.cpp]: alignment issue. when you fix the code, please see if it looks more readable in your laptop screen and then submit for review.

File: test/src/sender/main.cpp [hunk 22]
```diff
@@ -13,129 +13,151 @@
...
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
+        cadmium::dynamic::translate::make_EOC <sender_defs::packetSentOut,
+			output_pack> ("sender1"),
+        cadmium::dynamic::translate::make_EOC <sender_defs::ackReceivedOut,
+			output_acknowledgement> ("sender1"),
+        cadmium::dynamic::translate::make_EOC <sender_defs::dataOut, 
+			output_data> ("sender1")
+    };
+    cadmium::dynamic::modeling::ICs ics_TOP = {
```
Comment (syedomarazam) [inline test/src/sender/main.cpp]: alignment issue. looking like climbing steps.

File: test/src/subnet/main.cpp [hunk 23]
```diff
@@ -13,117 +13,131 @@
...
- oports_TOP, 
- eics_TOP, 
- eocs_TOP, 
- ics_TOP 
-  );
-
-///****************////
-
-    auto elapsed1 = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
+    using info = cadmium::logger::logger <cadmium::logger::logger_info,
```
Comment (syedomarazam) [inline test/src/subnet/main.cpp]: alignment issue in Line 55 to 73.

File: test/src/subnet/main.cpp [hunk 24]
```diff
@@ -13,117 +13,131 @@
...
+    /************************/
+    cadmium::dynamic::modeling::Ports iports_TOP = {};
+    cadmium::dynamic::modeling::Ports oports_TOP = {
+        typeid(output_port)
+    };
+    cadmium::dynamic::modeling::Models submodels_TOP = {
+        generator, subnet1
+    };
+    cadmium::dynamic::modeling::EICs eics_TOP = {};
+    cadmium::dynamic::modeling::EOCs eocs_TOP = {
```
Comment (syedomarazam) [inline test/src/subnet/main.cpp]: alignment issue

File: test/src/subnet/main.cpp [hunk 25]
```diff
@@ -13,117 +13,131 @@
...
+											cadmium::dynamic::translate::make_EOC <subnet_defs::out, output_port> ("subnet1")
+    };
+    cadmium::dynamic::modeling::ICs ics_TOP = {
```
Comment (syedomarazam) [inline test/src/subnet/main.cpp]: alignment issue

File: test/src/subnet/main.cpp [hunk 26]
```diff
@@ -13,117 +13,131 @@
...
+										cadmium::dynamic::translate::make_IC <iestream_input_defs 
+																					<message_t>::out, subnet_defs:: in> ("generator", "subnet1")
+    };
+    std::shared_ptr <cadmium::dynamic::modeling::coupled <TIME>> TOP = 
+		std::make_shared <cadmium::dynamic::modeling::coupled <TIME>> (
```
Comment (syedomarazam) [inline test/src/subnet/main.cpp]: alignment issue. must align to where the parenthesis is starting.

**Review by pcsmartapps — CHANGES_REQUESTED** [review]

**Review by syedomarazam — CHANGES_REQUESTED** [review]

File: include/atomics/sender_cadmium.hpp [hunk 27]
```diff
@@ -0,0 +1,184 @@
...
+};
+
+template <typename TIME>
+    class Sender
+    {
+        using definitions = sender_defs; // putting definitions in context
+        public:
+            //Parameters to be overwriten when instantiating the atomic model
+            TIME PREPARATION_TIME;
+			TIME timeout;
```
Comment (syedomarazam) [inline include/atomics/sender_cadmium.hpp]: timeout will change to TIME_OUT as you did for PREPARATION_TIME

File: test/src/sender/main_sender.cpp [hunk 28]
```diff
@@ -0,0 +1,161 @@
...
+    /********************************************/
+    /****** APPLICATION GENERATOR *******************/
+    /********************************************/
+    string input_data_control = "test/data/sender_input_test_control_In.txt"; 
+    const char *i_input_data_control = input_data_control.c_str();
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_con = 
+    cadmium::dynamic::translate::make_dynamic_atomic_model 
+    < ApplicationGen, TIME, const char *> ("generator_con", 
+																																				std::move(i_input_data_control));
```
Comment (syedomarazam) [inline test/src/sender/main_sender.cpp]: the line is out of the screen.

File: test/src/sender/main_sender.cpp [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
+
+    string input_data_ack = "test/data/sender_input_test_ack_In.txt"; 
+    const char *i_input_data_ack = input_data_ack.c_str();
+
+    std::shared_ptr <cadmium::dynamic::modeling::model> generator_ack = 
+    cadmium::dynamic::translate::make_dynamic_atomic_model 
+    < ApplicationGen, TIME, const char *> ("generator_ack", std::move 
+																																								(i_input_data_ack));
```
Comment (syedomarazam) [inline test/src/sender/main_sender.cpp]: the line is out of the screen.

File: include/atomics/sender_cadmium.hpp [hunk 30]
```diff
@@ -0,0 +1,184 @@
...
+#include <random>
+
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct sender_defs{
+	struct packetSentOut: public out_port <message_t> {};
```
Comment (syedomarazam) [inline include/atomics/sender_cadmium.hpp]: confirmed with professor just now that the members of the structure must be implemented per coding conventions.

like packet_sent_out;
       ack_received_out;

change for all variables inside this structure.

**Review by syedomarazam — CHANGES_REQUESTED** [review]
Approved files:
1) receiver_cadmium.hpp
2) subnet_cadmium.hpp

Changes required in files:

in sender_cadmium.hpp

1) structure members need to follow coding conventions. 
2) timeout is a constant, must be capitalized. Looks my line comments for specific changes.


in main_receiver.cpp, main_sender.cpp, main_subnet.cpp

1) Line < 80  character rule is implemented but code is not aligned properly. Code is less readable.
   Confirmed with the professor that it has to be more readable when we break a line into multiple.

**Review by syedomarazam — APPROVED** [review]
Code is implemented per good coding practices defined in the class. Developers have fixed all the review comments.

File: include/atomics/receiver_cadmium.hpp [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+            return bags;
+			}
+
+            // time_advance function
+            TIME time_advance() const {  
+				TIME next_internal;
+				if (state.sending) {
+                next_internal = PREPARATION_TIME;
+                }
+					else {
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp]: Too many indentations. Pls, see professor notes

File: include/atomics/receiver_cadmium.hpp [hunk 32]
```diff
@@ -0,0 +1,118 @@
...
+                next_internal = std::numeric_limits<TIME>::infinity();
+					}
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp:105]: else should be lined up with if.Pls, see professor notes

File: include/atomics/receiver_cadmium.hpp [hunk 33]
```diff
@@ -0,0 +1,118 @@
...
+                next_internal = PREPARATION_TIME;
+                }
+					else {
+                next_internal = std::numeric_limits<TIME>::infinity();
+					}    
+            return next_internal;
+            }
+
+            friend std::ostringstream& operator<<(std::ostringstream& os,
+												  const typename Receiver<TIME>::state_type& i) {
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp:110]: too many indentations in line 111

File: include/atomics/receiver_cadmium.hpp [hunk 34]
```diff
@@ -0,0 +1,118 @@
...
+					else {
+                next_internal = std::numeric_limits<TIME>::infinity();
+					}    
+            return next_internal;
+            }
+
+            friend std::ostringstream& operator<<(std::ostringstream& os,
+												  const typename Receiver<TIME>::state_type& i) {		
+                os << "acknowledgement_num: " << i.acknowledgement_num; 
+				return os;
```
Comment (pcsmartapps) [inline include/atomics/receiver_cadmium.hpp:112]: return os should be lined up properly .  Pls also check, subnet_cadmium.hpp

File: include/atomics/subnet_cadmium.hpp [hunk 35]
```diff
@@ -27,87 +27,96 @@
...
             // output function
             typename make_message_bags<output_ports>::type output() const {
-              typename make_message_bags<output_ports>::type bags;
-              Message_t out;
-              if ((double)rand() / (double) RAND_MAX  < 0.95){
-                out.value = state.packet;
-                get_messages<typename defs::out>(bags).push_back(out);
-              }
-              return bags;
+						typename make_message_bags<output_ports>::type bags;
```
Comment (pcsmartapps) [inline include/atomics/subnet_cadmium.hpp:90]: typename indented to much

**Review by pcsmartapps — CHANGES_REQUESTED** [review]

File: include/atomics/subnet_cadmium.hpp (hunk 35 above)
Comment (jrajeev459) [reply include/atomics/subnet_cadmium.hpp:90]: download the files and open the file in your local editor. you can observe them as aligned with coding conventions.

Comment (jrajeev459) [reply include/atomics/subnet_cadmium.hpp:90]: the same thing applies to all the comments. open them in local editor then you can observe them with coding standards

Comment (pcsmartapps) [reply include/atomics/subnet_cadmium.hpp:90]: I opened your code using sublime,notepad and VS 2015, the issues raised still stand.

**Review by mohamedali5 — APPROVED** [review]

**Review by pcsmartapps — APPROVED** [review]
Alignment issue under include/atomics folder have been resolved.

**Pull request merged**